### PR TITLE
New "Mugger" Spaceport mission

### PR DESCRIPTION
**Pull request by Anarchist2**
## Summary
Adds a new mission where the player gets held up and extorted for a free ride by someone whose preparedness is wanting.

This PR also makes some changes to "Transport Workers B".

## Save File
This save file can be used to play through the new mission content:
[I II II I-.txt](https://github.com/endless-sky/endless-sky/files/8356754/I.II.II.I-.txt)

### CODE REVIEW

**Comment by Anarchist2** [comment]
Marking as a draft since I want to rewrite some stuff.

**Comment by Zitchas** [comment]
I'm slightly dubious about that backflip actually working, but it certainly is entertaining. Nice little story.

**Comment by Anarchist2** [comment]
While the death options might work in a vacuum (or a semi-vacuum, considering FW Katya), when in the context of the rest of the mission, it ends up raising some issues that can't really be solved without writing beyond the intent of the mission (Most prominently being, "Why is she telling me her entire life story when she was about to kill me?" Even the current explanation of "She doesn't really have the guts to kill another person," feels borderline to me.).

File: data/human/human missions.txt [hunk 1]
```diff
@@ -3833,3 +3832,158 @@ mission "Spaceport Reminder"
...
+				`	(Backwards somersault over your attacker and disarm her.)`
+					goto flip
+			
+			`	You turn around as swiftly as you can as your hand grasps for your pistol. Due to surprise, or perhaps nervousness, the woman threatening you fails to fire the shotgun she's holding in the second you spend turning. You manage to land a shot, and your quarry falls while a cry escapes from her body. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
+				decline
+			
+			label flip
+			`	You push down as hard as you can with your legs while leaning backwards, rising to the lofty height of two feet above the ground. As you begin to fall back-first uncontrollably, you feel metal dig into your spine as a gasp escapes from behind you, before landing on a soft body. You pick yourself up as fast as possible, and you end up face-to-face with your attacker. In the darkness, you pick out that she's wearing a black hoodie with a bandanna over her mouth, her back hunched over and feet placed apart. As you look down, you spot a sawed-off breach-action shotgun held single-handed, the end of which is pointed at your abdomen. "Hands up now, Captain."`
```
Comment (Pointedstick) [inline data/human/human missions.txt:3876]:
```suggestion
			`	You push down as hard as you can with your legs while leaning backwards, rising to the lofty height of two feet above the ground. As you begin to fall back-first uncontrollably, you feel metal dig into your spine as a gasp escapes from behind you, before landing on a soft body. You pick yourself up as fast as possible, and you end up face-to-face with your attacker. In the darkness, you pick out that she's wearing a black hoodie with a bandanna over her mouth, her back hunched over and feet placed apart. As you look down, you spot a sawed-off break-action shotgun held single-handed, the end of which is pointed at your abdomen. "Hands up now, Captain."`
```

(there's no such thing as a "breach-action shotgun")

File: data/human/human missions.txt [hunk 2]
```diff
@@ -3833,3 +3832,158 @@ mission "Spaceport Reminder"
...
+			choice
+				`	(Put your hands up.)`
+					goto handsup
+				`	(Draw your gun and fire.)`
+			`	You hand darts for your gun, and you manage to aim and fire before the woman does. She collapses on the ground while a cry escapes from her body. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
+				decline
+			
+			label peaceful
+			`	You raise your arms and turn to face the woman. In the darkness, you pick out that she's wearing a black hoodie with a bandanna over her mouth, her back hunched over and feet placed apart. As you look down, you spot a sawed-off breach-action shotgun held single-handed, the end of which is pointed at your abdomen.`
```
Comment (Pointedstick) [inline data/human/human missions.txt:3885]:
```suggestion
			`	You raise your arms and turn to face the woman. In the darkness, you pick out that she's wearing a black hoodie with a bandanna over her mouth, her back hunched over and feet placed apart. As you look down, you spot a sawed-off break-action shotgun held single-handed, the end of which is pointed at your abdomen.`
```

File: data/human/human missions.txt [hunk 3]
```diff
@@ -3833,3 +3832,158 @@ mission "Spaceport Reminder"
...
+		has "event: fumbling mugger timer"
+	on offer
+		conversation
+			`Your exploration of the spaceport leads you to a dark alley where the number of working lights are far outnumbered by the number of broken ones. The contents of steel bins lining the walls overflow onto the pavement, attracting swarms of flies to them. On the surfaces that are illuminated, you can see a mess of black scrawlings upon bare, cracked concrete, and the few windows present on the towering walls surrounding you are either boarded up or shattered. A dark crimson stains the bottom half of a broken window.`
+			`	Suddenly, you feel a cold object pushed against your back as a woman speaks softly. "Hands where I can see them, Captain."`
+			choice
+				`	(Lift your hands up and slowly turn.)`
+					goto peaceful
+				`	(Spin around and fire.)`
+				`	(Backwards somersault over your attacker and disarm her.)`
```
Comment (Pointedstick) [inline data/human/human missions.txt:3869]: I might rewrite this in a slightly more plausible way. :) Like this:

```suggestion
				`	(Try a fancy acrobatic move and disarm your attacker.)`
```

File: data/human/human missions.txt [hunk 4]
```diff
@@ -3833,3 +3832,158 @@ mission "Spaceport Reminder"
...
+				`	(Try a fancy acrobatic move and disarm your attacker.)`
+					goto flip
+			
+			`	You turn around as swiftly as you can as your hand grasps for your pistol. Due to surprise, or perhaps nervousness, the woman threatening you fails to fire the shotgun she's holding in the second you spend turning. You manage to land a shot, and your quarry falls while a cry escapes from her body. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
+				decline
+			
+			label flip
+			`	You push down as hard as you can with your legs while leaning backwards, rising to the lofty height of two feet above the ground. As you begin to fall back-first uncontrollably, you feel metal dig into your spine as a gasp escapes from behind you, before landing on a soft body. You pick yourself up as fast as possible, and you end up face-to-face with your attacker. In the darkness, you pick out that she's wearing a black hoodie with a bandanna over her mouth, her back hunched over and feet placed apart. As you look down, you spot a sawed-off break-action shotgun held single-handed, the end of which is pointed at your abdomen. "Hands up now, Captain."`
```
Comment (W1zrad) [inline data/human/human missions.txt:3876]:
```suggestion
			`	You push down as hard as you can with your legs while leaning backwards, rising to the lofty height of two feet above the ground. As you begin to fall back-first uncontrollably, cold metal digs into your spine. A gasp escapes from behind as you land on a soft body. You pick yourself up as fast as possible, and you end up face-to-face with your attacker. In the darkness, you pick out that she's wearing a black hoodie with a bandanna over her mouth, her back hunched over and feet placed apart. As you look down, you spot a sawed-off break-action shotgun held single-handed, the end of which is pointed at your abdomen. "Hands up now, Captain."`
```
That `two feet` bit made me chuckle :)
I feel like that middle sentence could use some work - feels very long and messy as-is. If you've got any alternative ideas than the ones I'm suggesting, be my guest!

File: data/human/human missions.txt [hunk 5]
```diff
@@ -3833,3 +3832,158 @@ mission "Spaceport Reminder"
...
+			choice
+				`	(Put your hands up.)`
+					goto handsup
+				`	(Draw your gun and fire.)`
+			`	You hand darts for your gun, and you manage to aim and fire before the woman does. She collapses on the ground while a cry escapes from her body. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
```
Comment (W1zrad) [inline data/human/human missions.txt:3881]:
```suggestion
			`	You hand darts for your gun, and you somehow manage to aim and fire before the woman does. She collapses on the ground while a cry escapes from her body. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
```

File: data/human/human missions.txt [hunk 6]
```diff
@@ -3833,3 +3832,158 @@ mission "Spaceport Reminder"
...
+				`	(Threaten her with your gun.)`
+					goto threaten
+				`	(Shoot her.)`
+			
+			`	Before she can move a muscle, you whip out your pistol and fire, watching as the woman collapses onto the ground with a cry. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
+				decline
+			
+			label threaten
+			`	You take out your gun and aim it at the woman's head. "Drop the gun," you say, "or I'll shoot."`
+			`	She seizes up for a few moments, before throwing her shotgun away and raising her hands. "I'm, um, sorry about threatening you. Would you mind if the both of us just walked this off?"`
```
Comment (W1zrad) [inline data/human/human missions.txt:3916]:
```suggestion
			`	She seizes up for a few moments, then she throws her shotgun away and raises her hands. "I'm, um, sorry about threatening you. Would you mind if the both of us just walked this off?"`
```

**Review by W1zrad — CHANGES_REQUESTED** [review]
Looking solid to me. Seems like there's a follow-up planned for this?

File: data/human/human missions.txt [hunk 7]
```diff
@@ -3833,3 +3832,158 @@ mission "Spaceport Reminder"
...
+			label leave
+			`	She nods in response. "I'll try my best."`
+				decline
+			label gift
+			action
+				payment -15000
+				set "gifted money to mugger"
+				log `Was held up by a woman who wanted transport to Maker. Managed to gain the upper hand in the engagement, but decided to pay for her transport anyway.`
+			`	She is almost incredulous at your gift. "I really can't thank you enough for helping me out." The two of you say your farewells as she gets ready to depart your ship.`
+				decline
```
Comment (Amazinite) [inline data/human/human missions.txt:3979]: Don't need an endpoint at the end of a conversation that isn't in an on offer.
```suggestion
```

File: data/human/human missions.txt [hunk 8]
```diff
@@ -3833,3 +3832,158 @@ mission "Spaceport Reminder"
...
+			choice
+				`	(Wish her good luck.)`
+					goto leave
+				`	(Give her 15,000 credits before dropping her off.)`
+					goto gift
+			label poor
+			choice
+				`	(Wish her good luck.)`
+			label leave
+			`	She nods in response. "I'll try my best."`
```
Comment (Amazinite) [inline data/human/human missions.txt:3971]:
```suggestion
			`	She nods in response. "I'll try my best," she says solemnly as she prepares to depart your ship.`
```

**Review by Amazinite — CHANGES_REQUESTED** [review]
Bruh moment.

**Comment by Amazinite** [comment]
Might be worth setting a condition for the scenarios where you shoot.

File: data/human/human missions.txt [hunk 9]
```diff
@@ -3873,7 +3873,7 @@ mission "Fumbling Mugger"
...
+			`	You push down as hard as you can with your legs while leaning backwards, rising to the lofty height of one foot above the ground. As you begin to fall back-first uncontrollably, cold metal digs into your spine. A gasp escapes from behind as you land on a soft body. You pick yourself up as fast as possible, and you end up face-to-face with your attacker. In the darkness, you pick out that she's wearing a black hoodie with a bandanna over her mouth, her back hunched over and feet placed apart. As you look down, you spot a sawed-off break-action shotgun held single-handed, the end of which is pointed at your abdomen. "Hands up now, Captain."`
```
Comment (Amazinite) [inline data/human/human missions.txt:3876]: I feel like there's something more comedic about two feet than one foot, so I'd flip it back.

File: data/human/human missions.txt [hunk 10]
```diff
@@ -3833,3 +3832,161 @@ mission "Spaceport Reminder"
...
+				decline
+			
+			label flip
+			`	You push down as hard as you can with your legs while leaning backwards, rising to the lofty height of one foot above the ground. As you begin to fall back-first uncontrollably, cold metal digs into your spine. A gasp escapes from behind as you land on a soft body. You pick yourself up as fast as possible, and you end up face-to-face with your attacker. In the darkness, you pick out that she's wearing a black hoodie with a bandanna over her mouth, her back hunched over and feet placed apart. As you look down, you spot a sawed-off break-action shotgun held single-handed, the end of which is pointed at your abdomen. "Hands up now, Captain."`
+			choice
+				`	(Put your hands up.)`
+					goto handsup
+				`	(Draw your gun and fire.)`
+			`	You hand darts for your gun, and you somehow manage to aim and fire before the woman does. She collapses on the ground while a cry escapes from her body. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
```
Comment (Amazinite) [inline data/human/human missions.txt:3881]:
```suggestion
			action
				set "fumbling mugger shot"
			`	You hand darts for your gun, and you somehow manage to aim and fire before the woman does. She collapses on the ground while a cry escapes from her body. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
```

File: data/human/human missions.txt [hunk 11]
```diff
@@ -3833,3 +3832,161 @@ mission "Spaceport Reminder"
...
+			`	Suddenly, you feel a cold object pushed against your back as a woman speaks softly. "Hands where I can see them, Captain."`
+			choice
+				`	(Lift your hands up and slowly turn.)`
+					goto peaceful
+				`	(Spin around and fire.)`
+				`	(Try a fancy acrobatic move and disarm your attacker.)`
+					goto flip
+			
+			`	You turn around as swiftly as you can as your hand grasps for your pistol. Due to surprise, or perhaps nervousness, the woman threatening you fails to fire the shotgun she's holding in the second you spend turning. You manage to land a shot, and your quarry falls while a cry escapes from her body. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
```
Comment (Amazinite) [inline data/human/human missions.txt:3872]:
```suggestion
			action
				set "fumbling mugger shot"
			`	You turn around as swiftly as you can as your hand grasps for your pistol. Due to surprise, or perhaps nervousness, the woman threatening you fails to fire the shotgun she's holding in the second you spend turning. You manage to land a shot, and your quarry falls while a cry escapes from her body. As you walk away, her body fades into the darkness, blending with the garbage bags around it.`
```

**Review by Amazinite — CHANGES_REQUESTED** [review]

**Review by Amazinite — APPROVED** [review]

**Review by W1zrad — APPROVED** [review]

**Pull request merged**